sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryCompilationErrors.scala

```diff
@@ -1349,9 +1349,13 @@ private[spark] object QueryCompilationErrors {
         s"${evalTypes.mkString(",")}")
   }
 
-  def ambiguousFieldNameError(fieldName: String, names: String): Throwable = {
+  def ambiguousFieldNameError(
+      fieldName: Seq[String], nunMatches: Int, context: Origin): Throwable = {
     new AnalysisException(
-      s"Ambiguous field name: $fieldName. Found multiple columns that can match: $names")
+      errorClass = "INVALID_FIELD_NAME",
+      messageParameters = Array(fieldName.quoted,
+        s"${fieldName.last} is ambiguous and has $nunMatches matching fields in the struct"),
```

Comment:
Can we put the majority of this in the error message format? Eg. the format would be `Field name %s is invalid: %s is ambiguous and has %s matching fields in the struct`, and the parameters would be `Array(fieldName.quoted, fieldName.last, numMatches)`

Reply:
In this case, we would have to create two new error classes (another one for the case that path is not a struct).